src/EvoSC.Common/Interfaces/Services/IChatService.cs

```diff
@@ -0,0 +1,225 @@
+﻿using EvoSC.Common.Interfaces.Models;
+using EvoSC.Common.Util.TextFormatting;
+
+namespace EvoSC.Common.Interfaces.Services;
+
+public interface IChatService
+{
+    /// <summary>
+    /// Send a chat message to all players.
+    /// </summary>
+    /// <param name="message">Message to send</param>
+    /// <returns></returns>
+    public Task<bool> SendChatMessageAsync(string message);
+    /// <summary>
+    /// Send a chat message to all players.
+    /// </summary>
+    /// <param name="message">Message to send</param>
+    /// <returns></returns>
+    public Task<bool> SendChatMessageAsync(TextFormatter message);
+    /// <summary>
+    /// Send a chat message to all players.
+    /// </summary>
+    /// <param name="message">Message to send</param>
+    /// <returns></returns>
+    public Task<bool> SendChatMessageAsync(Action<TextFormatter> message);
+    /// <summary>
+    /// Send a chat message to a set of players.
+    /// </summary>
+    /// <param name="message">Message to send.</param>
+    /// <param name="accountIds">Players to send the message to.</param>
+    /// <returns></returns>
+    public Task<bool> SendChatMessageAsync(string message, params string[] accountIds);
+    /// <summary>
+    /// Send a chat message to a set of players.
+    /// </summary>
+    /// <param name="message">Message to send.</param>
+    /// <param name="accountIds">Players to send the message to.</param>
+    /// <returns></returns>
+    public Task<bool> SendChatMessageAsync(string message, params IPlayer[] players);
+    /// <summary>
+    /// Send a chat message to a set of players.
+    /// </summary>
+    /// <param name="message">Message to send.</param>
+    /// <param name="accountIds">Players to send the message to.</param>
+    /// <returns></returns>
+    public Task<bool> SendChatMessageAsync(TextFormatter message, params string[] accountIds);
+    /// <summary>
+    /// Send a chat message to a set of players.
+    /// </summary>
+    /// <param name="message">Message to send.</param>
+    /// <param name="accountIds">Players to send the message to.</param>
+    /// <returns></returns>
+    public Task<bool> SendChatMessageAsync(TextFormatter message, params IPlayer[] players);
+    /// <summary>
+    /// Send a chat message to a set of players.
+    /// </summary>
+    /// <param name="message">Message to send.</param>
+    /// <param name="accountIds">Players to send the message to.</param>
+    /// <returns></returns>
+    public Task<bool> SendChatMessageAsync(Action<TextFormatter> message, params string[] accountIds);
```

Comment:
Could be useful with a few words on the difference between Action and without action. Same applies to line 19/25